projects/train/train/data/supervised/frequency_domain.py

```diff
@@ -23,14 +23,73 @@ def setup(self, *args, **kwargs):
         )
 
     def augment(self, X):
-        X, y = super().augment(X)
+        X, y, psds = super().augment(X)
+        X = self.whitener(X, psds)
         X = self.qtransform(X)
         return X, y
 
     def build_val_batches(
         self, background: Tensor, signals: Tensor
     ) -> tuple[Tensor, Tensor]:
-        X_bg, X_fg = super().build_val_batches(background, signals)
+        X_bg, X_inj, psds = super().build_val_batches(background, signals)
+
+        # whiten and q transform background
+        X_bg = self.whitener(X_bg, psds)
         X_bg = self.qtransform(X_bg)
-        X_fg = torch.stack([self.qtransform(X) for X in X_fg])
+
+        # whiten and q transform each view of injections
+        X_fg = []
+        for inj in X_inj:
+            inj = self.whitener(inj, psds)
+            X_fg.append(self.qtransform(inj))
+
+        X_fg = torch.stack(X_fg)
         return X_bg, X_fg
+
+
+class FrequencyDomainAframeDataset(SupervisedAframeDataset):
+    def build_transforms(self, *args, **kwargs):
+        super().build_transforms(*args, **kwargs)
+
+        # build tapering window
+        window_length = self.hparams.kernel_length + self.hparams.fduration
+        window_size = int(window_length * self.sample_rate)
+        window = torch.hann_window(window_size)
+        df = 1 / window_length
+        window_scale = torch.sum(window**2) / len(window)
+        window /= torch.sqrt(window_scale)
+        window *= torch.sqrt(torch.tensor(4 * df))
+        self.window = window[None][None].to(self.device)
```

Comment:
Can you explain or point to something that explains why the window is constructed this way?

Reply:
Yeah this is to ensure that the fft has unit variance mean 0 after the fft - the window takes away some power, so you have to divide by this window factor